Sources/Core/Tracker/Tracker.swift

```diff
@@ -40,8 +40,8 @@ func uncaughtExceptionHandler(_ exception: NSException) {
 class Tracker: NSObject {
     private var platformContextSchema: String = ""
     private var dataCollection = true
-
     private var builderFinished = false
+    private let eventProcessQueue = DispatchQueue(label: "snowplow.track")
```

Comment:
Could we remove all of the `objc_sync_enter()` and exits from the setters in this `Tracker` file ([such as here](https://github.com/snowplow/snowplow-ios-tracker/blob/8a3d0d6dc19eb8b86570080db85e7e3775013141/Sources/Core/Tracker/Tracker.swift#L178)) and instead dispatch them to this queue (I think async dispatch would be better than sync)?